File: src/main/java/tech/zolhungaj/amqapi/servercommands/expandlibrary/ExpandLibraryEntryUpdated.java
package tech.zolhungaj.amqapi.servercommands.expandlibrary;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;

import java.util.Optional;

public record ExpandLibraryEntryUpdated(
	int annId,
	int annSongId,
	String host,
	int resolution,
	@Json(name = "animeStillAvaliable") Boolean animeStillAvailable,
	@Json(name = "songStillAvaliable") Optional<Boolean> songStillAvailable
) implements Command {
	@Override
	public String commandName() {
		return CommandTypeOld.EXPAND_LIBRARY_UPDATE.commandName;
	}
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/expandlibrary/ExpandLibraryEntryUpdated.kt
package tech.zolhungaj.amqapi.servercommands.expandlibrary

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType

@JvmRecord
@CommandType("expandLibrary song answered")
data class ExpandLibraryEntryUpdated(
    @Json(name = "annId")
    val animeNewsNetworkId: Int,
    @Json(name = "annSongId")
    val songId : Int,
    val host: String,
    val resolution: Int,
    @Json(name = "animeStillAvaliable")
    val animeStillAvailable : Boolean,
    @Json(name = "songStillAvaliable")
    val songStillAvailable : Boolean?
)
